core/src/main/java/org/apache/calcite/runtime/SqlFunctions.java

```diff
@@ -1854,6 +1854,20 @@ public static double acos(double b0) {
     return Math.acos(b0);
   }
 
+  // ACOSH
+  /** SQL <code>ACOSH</code> operator applied to BigDecimal values. */
+  public static double acosh(BigDecimal b0) {
+    return acosh(b0.doubleValue());
+  }
+
+  /** SQL <code>ACOSH</code> operator applied to double values. */
+  public static double acosh(double b0) {
+    if (b0 < 1) {
+      throw new IllegalArgumentException("Input parameter of acosh cannot be less than 1!");
+    }
+    return Math.log(Math.sqrt(b0 * b0 - 1.0) + b0);
```

Comment:
Super-nitpick: for consistency with the other methods, I think we should use `1.0d` instead of  `1.0` here